agents/official/beeai-supervisor/src/supervisor/supervisor.ts

```diff
@@ -0,0 +1,116 @@
+import { AcpServer } from "@i-am-bee/acp-sdk/server/acp.js";
+import { MessageOutput } from "@i-am-bee/beeai-sdk/schemas/message";
+import { Metadata } from "@i-am-bee/beeai-sdk/schemas/metadata";
+import {
+  textInputSchema,
+  textOutputSchema,
+} from "@i-am-bee/beeai-sdk/schemas/text";
+import {
+  createRuntime as createSupervisor,
+  RuntimeOutputMethod,
+} from "@i-am-bee/beeai-supervisor";
+import { CreateAgentConfig } from "@i-am-bee/beeai-supervisor/agents/registry/registry.js";
+import { z } from "zod";
+import { AgentFactory } from "./agent-factory.js";
+import { PlatformSdk } from "./platform-sdk.js";
+
+const inputSchema = textInputSchema.extend({
+  availableAgents: z.array(z.string()),
+});
+type Input = z.infer<typeof inputSchema>;
+const outputSchema = textOutputSchema;
+
+const run =
+  (server: AcpServer) =>
+  async (
+    {
+      params: { input, _meta },
+    }: {
+      params: {
+        input: Input;
+        _meta?: { progressToken?: string | number };
+      };
+    },
+    { signal }: { signal?: AbortSignal }
+  ) => {
+    // ****************************************************************************************************
+    // Connect platform
+    // ****************************************************************************************************
+    const platformSdk = PlatformSdk.getInstance();
+    await platformSdk.init(
+      input.availableAgents.map((a) => a.toLocaleLowerCase())
+    );
+    const listedPlatformAgents = await platformSdk.listAgents();
+    const agentConfigFixtures = listedPlatformAgents.map(
+      ({ beeAiAgentId, description }) =>
+        ({
+          agentKind: "operator",
+          agentConfigVersion: 1,
+          agentType: beeAiAgentId,
+          description: description,
+          autoPopulatePool: false,
+          instructions: "Not used",
+          tools: [],
+          maxPoolSize: 10,
+        } as CreateAgentConfig)
+    );
+
+    const supervisorAgent = await createSupervisor({
+      agentConfigFixtures,
+      agentFactory: new AgentFactory(),
+      switches: {
+        agentRegistry: {
+          mutableAgentConfigs: false,
+          restoration: false,
+        },
+        taskManager: {
+          restoration: false,
+        },
+      },
+      workspace: "beeai",
+      outputDirPath: "./output",
+    });
+
+    const output: RuntimeOutputMethod = async (output) => {
+      let role;
+      if (output.agent) {
+        role = `🤖 [${output.agent.agentId}] `;
+      } else {
+        role = `📋 [${output.taskRun.taskRunId}] `;
+      }
+
+      if (_meta?.progressToken) {
+        await server.server.sendAgentRunProgress({
+          progressToken: _meta.progressToken,
+          delta: {
+            text: `${role} ${output.text}\n\n`,
+          },
+        });
+      }
+    };
+
+    const response = await supervisorAgent.run(input.text, output);
+    return {
+      text: response,
+      logs: [],
+    };
+  };
+
+export const agent = {
```

Comment:
I think this is incosistent with the other agents we have please, see others @aleskalfas